custom_components/plugwise/gateway.py

```diff
@@ -176,21 +176,24 @@ def __init__(self, coordinator, dev_id, name, model, vendor, fw):
         """Initialise the gateway."""
         super().__init__(coordinator)
 
-        self._attr_available = super().available
+        via_device = None
         entry = coordinator.config_entry
         gw_id = coordinator.data[0]["gateway_id"]
+        if dev_id != gw_id:
+            dev_name = name
+            via_device = (DOMAIN, gw_id)
```

Comment:
If you create/set `self._attr_device_info` first, you can at this point just add to it.
Makes it cleaner with less code.

e.g.:
```py
        self._attr_device_info = DeviceInfo(
            configuration_url=f"http://{entry.data[CONF_HOST]}",
            identifiers={(DOMAIN, dev_id)},
            name=dev_name = f"Smile {coordinator.data[0]['smile_name']}",
            manufacturer=vendor,
            model=model,
            sw_version=fw,
        )

        # Comment what this does
        if dev_id != gw_id:
            self._attr_device_info.update(
            name=name,
            via_device=(domain, gwid),
        )
```

Otherwise ✅

Reply:
Yes, of course, it's a dict. Thanks!